image/image_linux.go

```diff
@@ -890,7 +889,9 @@ var setupSeed = func(tsto *tooling.ToolingStore, model *asserts.Model, opts *Opt
 				return nil, err
 			}
 			if !sn.Info.Revision.Unset() {
-				imageManifest[sn.Info.SnapName()] = sn.Info.Revision
+				if err := s.manifest.MarkSnapRevisionSeeded(sn.Info.SnapName(), sn.Info.Revision.N); err != nil {
+					return nil, err
```

Comment:
do we have a test reaching here?

Reply:
We did not! I've added one in https://github.com/snapcore/snapd/pull/12579/commits/cc43d06fa86e98b1eef790827e843ca99ebf9916